packages/oauth-client/tests/test-CognitoClient.js

```diff
@@ -0,0 +1,373 @@
+const nock = require('nock');
+const test = require('ava');
+const cryptoRandomString = require('crypto-random-string');
+
+const { CognitoClient, CognitoError } = require('../dist');
+
+const randomString = () => cryptoRandomString({ length: 6 });
+
+const randomId = (prefix, separator = '-') =>
+  [prefix, randomString()].filter((x) => x).join(separator);
+
+const randomUrl = () => `http://${randomString()}.local`;
+
+const buildCognitoClient = () =>
+  new CognitoClient({
+    clientId: randomId('client-id'),
+    clientPassword: randomId('client-password'),
+    loginUrl: randomUrl(),
+    redirectUri: randomUrl(),
+  });
+
+const nockCognitoGet = (params) => {
+  const {
+    cognitoClient,
+    path,
+    requestHeaders = {},
+    responseStatus,
+    responseBody,
+  } = params;
+
+  return nock(
+    cognitoClient.loginUrl,
+    { reqheaders: requestHeaders }
+  )
+    .get(path)
+    .reply(responseStatus, responseBody);
+};
+
+const nockAuthPost = (params) => {
+  const {
+    cognitoClient,
+    path,
+    requestBody,
+    requestHeaders = {},
+    responseStatus,
+    responseBody,
+  } = params;
+
+  return nock(
+    cognitoClient.loginUrl,
+    { reqheaders: requestHeaders }
+  )
+    .post(path, requestBody)
+    .basicAuth({
+      user: cognitoClient.clientId,
+      pass: cognitoClient.clientPassword,
+    })
+    .reply(responseStatus, responseBody);
+};
+
+test.before(() => {
+  nock.disableNetConnect();
+});
+
+test('The CognitoClient constructor throws a TypeError if clientId is not specified', (t) => {
```

Comment:
test-OAuthClient has these TypeError tests and some other functions , no added value to tests in all three test files

Reply:
Yup, good call. I removed the duplicate constructor tests but left tests for `refreshAccessToken()` and `getAccessToken()` in both the test-CognitoClient and test-EarthdataLoginClient. Those two functions are in the parent class but we handle errors slightly differently and that's what those tests are looking for.

https://github.com/nasa/cumulus/pull/2209/commits/de8c1308412e3003874eef5be6078f16bef6e978